src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/ptv/ServiceChannelsTestsIT.java

```diff
@@ -0,0 +1,756 @@
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    
+    assertFound(String.format("/electronicServiceChannels/%s", electronicChannelId));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/electronicServiceChannels/%s", malformedId));
+    }
+    
+    assertNotFound(String.format("/electronicServiceChannels/%s", phoneChannelId));
+  }
+  
+  @Test
+  public void testPhoneChannelNotFound() throws InterruptedException {
+    String electronicChannelId = getElectronicChannelId(0);
+    String phoneChannelId = getPhoneChannelId(0);
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    
+    assertFound(String.format("/phoneServiceChannels/%s", phoneChannelId));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/phoneServiceChannels/%s", malformedId));
+    }
+    
+    assertNotFound(String.format("/phoneServiceChannels/%s", electronicChannelId));
+  }
+  
+  @Test
+  public void testPrintableFormChannelNotFound() throws InterruptedException {
+    String electronicChannelId = getElectronicChannelId(0);
+    String printableFormChannelId = getPrintableFormChannelId(0);
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    
+    assertFound(String.format("/printableFormServiceChannels/%s", printableFormChannelId));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/printableFormServiceChannels/%s", malformedId));
+    }
+    
+    assertNotFound(String.format("/printableFormServiceChannels/%s", electronicChannelId));
+  }
+
+  @Test
+  public void testServiceLocationChannelNotFound() throws InterruptedException {
+    String electronicChannelId = getElectronicChannelId(0);
+    String serviceLocationChannelId = getServiceLocationChannelId(0);
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    
+    assertFound(String.format("/serviceLocationServiceChannels/%s", serviceLocationChannelId));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/serviceLocationServiceChannels/%s", malformedId));
+    }
+    
+    assertNotFound(String.format("/serviceLocationServiceChannels/%s", electronicChannelId));
+  }
+
+  @Test
+  public void testWebPageChannelNotFound() throws InterruptedException {
+    String electronicChannelId = getElectronicChannelId(0);
+    String webPageChannelId = getWebPageChannelId(0);
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    
+    assertFound(String.format("/webPageServiceChannels/%s", webPageChannelId));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/webPageServiceChannels/%s", malformedId));
+    }
+    
+    assertNotFound(String.format("/webPageServiceChannels/%s", electronicChannelId));
+  }
+  
+  // TODO : Service REMOVE tests
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)